yarn-project/p2p/src/services/libp2p/libp2p_service.ts

```diff
@@ -751,6 +785,43 @@ export class LibP2PService<T extends P2PClientType = P2PClientType.Full> extends
     );
   }
 
+  private async extractTxsFromBlockProposal(block: BlockProposal, peerId: PeerId): Promise<void> {
+    const receivedTxs = block.txs ?? [];
+    const receivedTxHashes = await Promise.all(receivedTxs.map(tx => tx.getTxHash()));
+    if (receivedTxHashes.length === 0) {
+      return;
+    }
+    // Check if we already have the transactions in our pool
+    const haveAlready = await this.mempools.txPool.hasTxs(receivedTxHashes);
+    const newTxs = receivedTxs.filter((_, i) => !haveAlready[i]);
+    if (newTxs.length === 0) {
+      return;
+    }
+    // Validate the transactions
+    const blockNumber = (await this.archiver.getBlockNumber()) + 1;
+    const messageValidators = await this.createMessageValidators(blockNumber);
+    const validationResults = await Promise.all(
+      newTxs.map(tx => this.executeValidatorsOnTransaction(tx, messageValidators, peerId, blockNumber)),
+    );
+    const validTxs = newTxs.filter((_, i) => validationResults[i]);
+    if (validTxs.length === 0) {
+      return;
+    }
+    this.logger.verbose(`Adding ${validTxs.length} new txs to the tx pool, extracted from a block proposal.`);
```

Comment:
Let's log here:
- All the tx hashes received
- All the new tx hashes we found
- All the valid new tx hashes we are adding